src/validation.cpp

```diff
@@ -2611,7 +2647,19 @@ bool Chainstate::ConnectBlock(const CBlock& block, BlockValidationState& state,
         if (i > 0) {
             blockundo.vtxundo.emplace_back();
         }
-        UpdateCoins(tx, view, i == 0 ? undoDummy : blockundo.vtxundo.back(), pindex->nHeight);
+        if (swiftsync_active) {
+            UpdateCoinsWithHints(tx, view, *pindex, m_swiftsync_ctx.m_aggregate, hints.value());
+        } else {
+            UpdateCoins(tx, view, i == 0 ? undoDummy : blockundo.vtxundo.back(), pindex->nHeight);
+        }
+    }
+    if (swiftsync_active && m_swiftsync_ctx.StopHeight() == (uint32_t)pindex->nHeight) {
+        m_swiftsync_ctx.Completed();
+        if (m_swiftsync_ctx.m_aggregate.IsZero()) {
```

Comment:
`IsZero` requires context to understand why it means "valid."
What we're checking is: do all created outputs equal all spent inputs.
Consider tracking created and spent sums separately, then comparing for equality:
```
bool IsBalanced() const { return m_created == m_spent; }
```
This makes the invariant self-evident: all created outputs must equal all spent inputs. The extra 256 bits of state is negligible for the clarity gain.